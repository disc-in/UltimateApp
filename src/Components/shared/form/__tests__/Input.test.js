import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { Formik } from 'formik';
import * as Yup from 'yup';
import { TouchableOpacity } from 'react-native';

import Input from '../Input';

describe('Input', () => {
  it('renders correctly', () => {
    const { toJSON } = render(
      <Formik initialValues={{ title: 'My title' }}>
        <Input fieldName="title" label="Title" />
      </Formik>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly when required', () => {
    const { toJSON } = render(
      <Formik initialValues={{ title: 'My title' }}>
        <Input fieldName="title" label="Title" required />
      </Formik>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly when multiline', () => {
    const { toJSON } = render(
      <Formik initialValues={{ title: 'My title' }}>
        <Input fieldName="title" label="Title" multiline />
      </Formik>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly when there is an error', async () => {
    const { toJSON, getByTestId, getByDisplayValue, getByText } = render(
      <Formik
        initialValues={{ title: 'My title' }}
        validationSchema={Yup.object({ title: Yup.string().required().min(1000, 'Invalid title') })}
      >
        {({ handleSubmit }) => (
          <>
            <Input fieldName="title" label="Title" />
            <TouchableOpacity onPress={handleSubmit} testID="submit" />
          </>
        )}
      </Formik>,
    );

    expect(getByDisplayValue('My title')).toBeDefined();

    fireEvent.changeText(getByTestId('input-title'), 'new title');

    expect(getByDisplayValue('new title')).toBeDefined();

    await waitFor(() => {
      fireEvent.press(getByTestId('submit'));
    });

    expect(getByText('Invalid title')).not.toBeNull();
    expect(toJSON()).toMatchSnapshot();
  });
});
