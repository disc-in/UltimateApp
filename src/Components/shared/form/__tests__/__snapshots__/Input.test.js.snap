// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 10,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 5,
      }
    }
  >
    <Text>
      Title
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "borderColor": "#e5e5e5",
          "borderRadius": 5,
          "borderWidth": 0.5,
          "padding": 5,
        },
        undefined,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      rejectResponderTermination={true}
      testID="input-title"
      underlineColorAndroid="transparent"
      value="My title"
    />
  </View>
</View>
`;

exports[`Input renders correctly when multiline 1`] = `
<View
  style={
    Object {
      "marginBottom": 10,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 5,
      }
    }
  >
    <Text>
      Title
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "borderColor": "#e5e5e5",
          "borderRadius": 5,
          "borderWidth": 0.5,
          "padding": 5,
        },
        undefined,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      multiline={true}
      numberOfLines={3}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      rejectResponderTermination={true}
      testID="input-title"
      textAlignVertical="top"
      underlineColorAndroid="transparent"
      value="My title"
    />
  </View>
</View>
`;

exports[`Input renders correctly when required 1`] = `
<View
  style={
    Object {
      "marginBottom": 10,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "red",
          "fontStyle": "italic",
        }
      }
    >
      * 
    </Text>
    <Text>
      Title
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "borderColor": "#e5e5e5",
          "borderRadius": 5,
          "borderWidth": 0.5,
          "padding": 5,
        },
        undefined,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      rejectResponderTermination={true}
      testID="input-title"
      underlineColorAndroid="transparent"
      value="My title"
    />
  </View>
</View>
`;

exports[`Input renders correctly when there is an error 1`] = `
Array [
  <View
    style={
      Object {
        "marginBottom": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 5,
        }
      }
    >
      <Text>
        Title
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "borderColor": "#e5e5e5",
            "borderRadius": 5,
            "borderWidth": 0.5,
            "padding": 5,
          },
          Object {
            "borderColor": "red",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "red",
            "fontStyle": "italic",
          }
        }
      >
        Invalid title
      </Text>
      <TextInput
        allowFontScaling={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        rejectResponderTermination={true}
        testID="input-title"
        underlineColorAndroid="transparent"
        value="new title"
      />
    </View>
  </View>,
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="submit"
  />,
]
`;
